client/src/pages/AdvancedSearch/AdvancedSearch.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { useForm } from "./useForm";
+import {
+  AdvancedSearchWrapper,
+  Keyword,
+  Location,
+  Price,
+  Open,
+  Distance,
+  Form,
+} from "./AdvancedSearchStyles";
+import price_icon from "../../assets/images/price_icon.png";
+
+const AdvancedSearch = () => {
+  // getting the event handlers from our custom hook
+  const { onChange, onSubmit, values } = useForm(advancedSearchCallback);
+  // a submit function that will execute upon form submission
+  async function advancedSearchCallback() {
+    console.log(values);
+  }
+
+  return (
+    <AdvancedSearchWrapper>
+      <Form onSubmit={onSubmit}>
+        <fieldset>
+          <legend>
+            <h1>AdvancedSearch</h1>
+          </legend>
+          <Keyword>
+            <h2>Keyword</h2>
+            <input
+              name="keyword"
+              className="keyword-input"
+              onChange={onChange}
+              placeholder="vegan, gluten-free, etc."
+            />
+          </Keyword>
+
+          <Location>
+            <h2>Location</h2>
+            <input
+              name="location"
+              className="location-input"
+              onChange={onChange}
+              placeholder="city, state, zip"
+            />
+            <input
+              type="radio"
+              onChange={onChange}
+              id="current-radio"
+              value="current location data"
+            />
+            <label htmlFor="current-radio">Use Current</label>
+          </Location>
+
+          <Distance>
+            <h2>Distance</h2>
+            <input
+              onChange={onChange}
+              type="radio"
+              name="distance"
+              id="one"
+              value={1}
+            />
+            <label htmlFor="one">1</label>
+            <input
+              onChange={onChange}
+              type="radio"
+              name="distance"
+              id="five"
+              value={5}
+            />
+            <label htmlFor="five">5</label>
+            <input
+              onChange={onChange}
+              type="radio"
+              name="distance"
+              id="ten"
+              value={10}
+            />
+            <label htmlFor="ten">10</label>
+            <input
+              onChange={onChange}
+              type="radio"
+              name="distance"
+              id="twenty"
+              value={20}
+            />
+            <label htmlFor="twenty">20</label>
```

Comment:
I would say 5,10 & 20 would be fine, not sure about the 1, probably not needed. I you wanted 4 options, then you could add maybe 30?